command.go

```diff
@@ -919,6 +953,93 @@ func (cmd *StringSliceCmd) readReply(rd *proto.Reader) error {
 
 //------------------------------------------------------------------------------
 
+type KeyValue struct {
+	Key   string
+	Value string
+}
+
+type SliceKeyValueCmd struct {
+	baseCmd
+
+	val []KeyValue
+}
+
+var _ Cmder = (*SliceKeyValueCmd)(nil)
+
+func NewSliceKeyValueCmd(ctx context.Context, args ...interface{}) *SliceKeyValueCmd {
+	return &SliceKeyValueCmd{
+		baseCmd: baseCmd{
+			ctx:  ctx,
+			args: args,
+		},
+	}
+}
+
+func (cmd *SliceKeyValueCmd) Val() []KeyValue {
+	return cmd.val
+}
+
+func (cmd *SliceKeyValueCmd) Result() ([]KeyValue, error) {
+	return cmd.val, cmd.err
+}
+
+func (cmd *SliceKeyValueCmd) String() string {
+	return cmdString(cmd, cmd.val)
+}
+
+func (cmd *SliceKeyValueCmd) readReply(rd *proto.Reader) error { // nolint: dupl
+	_, err := rd.ReadArrayReply(func(rd *proto.Reader, n int64) (interface{}, error) {
+		if n == 0 {
+			cmd.val = make([]KeyValue, 0)
+			return nil, nil
+		}
+
+		b, err := rd.NextType()
```

Comment:
I would expect something like `rd.ReadReplyType`. `Next` implies that this is an iterator and `Type` and is not specific enough.